crates/devices/virtio-vsock/README.md

```diff
@@ -0,0 +1,120 @@
+# virtio-vsock
+
+The `virtio-vsock` crate provides abstractions for the components of the
+[vsock device](https://docs.oasis-open.org/virtio/virtio/v1.1/csprd01/virtio-v1.1-csprd01.html#x1-39000010).
+For now, it offers only an implementation for the vsock packet.
+The support is provided only for stream sockets.
+
+The vsock device is a socket device that can be used as a communication
+mechanism between the host and the guest. It is implemented using the virtio
+standard. The vsock device has three queues: an RX one, a TX one and an event
+one. In a simplified manner, the communication between the host and the guest
+is realized with the buffers that are exchanged using the device’s queues.
+These buffers are called packets in the vsock device context.
+
+## Vsock Packet
+
+### Design
+
+The virtio vsock packet is defined in the standard as having a header of type
+`virtio_vsock_hdr` and an optional `data` array of bytes. There are multiple
+operations that can be requested within a packet, e.g. `VIRTIO_VSOCK_OP_RST`
+for resetting the connection, `VIRTIO_VSOCK_OP_RW` for sending payload. Most
+operations are of the `VIRTIO_VSOCK_OP_RW` type, which means for data transfer,
+and the other ones are used for connection and buffer space management.
+`data` is non-empty only for the `VIRTIO_VSOCK_OP_RW` operations.
+
+The abstraction used for the packet implementation is the `VsockPacket`.
+It is using
+[`VolatileSlice`](https://github.com/rust-vmm/vm-memory/blob/fc7153a4f63c352d1fa9419c4654a6c9aec408cb/src/volatile_memory.rs#L266)s
```

Comment:
Can you also add a comment here about why we choose to go with VolatileSlice?

Reply:
Done.